web/source/keymanweb.js

```diff
@@ -3206,16 +3191,35 @@ if(!window['keyman']['initialized']) {
       return true;
     }                
 
-    keymanweb.callKeyboardStartGroup = function(Ltarg, Levent) {
+    /**
+     * Function     processKeystroke
+     * Scope        Private
+     * @param       {Object}        device      The device object properties to be utilized for this keystroke.
+     * @param       {Object}        element     The page element receiving input
+     * @param       {Object}        keystroke   The input keystroke (with its properties) to be mapped by the keyboard.
+     * Description  Encapsulates calls to keyboard input processing.
+     * @returns     {number}        0 if no match is made, otherwise 1.
+     */
+    keymanweb.processKeystroke = function(device, element, keystroke) {
```

Comment:
Given we are resetting cached selection, deadkey matches and cached context here, does that cause any issues with the new callers to the function? I'm not familiar enough with the old codepaths to be sure that the behaviour will be the same.

Reply:
I tested with the French keyboard for deadkey matches.  As for cached context, that appears to be largely for dealing with `any`, `index`, and/or `context` statements.  It's simply cleaning house before rounds of processing.